implemented a new feature that Hashie::Mash.load accepts a Pathname object #331

gipcompany opened the pull request:
As suggested on https://github.com/intridea/hashie/issues/331, I implemented the feature so that Hashie::Mash can also load a Pathname object.

dblock commented on lib/hashie/extensions/parsers/yaml_erb_parser.rb:
```diff
@@ -11,6 +11,7 @@ def initialize(file_path)
 
         def perform
           template = ERB.new(@content)
+          @file_path = @file_path.to_s if @file_path.respond_to? :to_s
```
I don't think this is the right place to do this, if `perform` is called over and over again it's wasteful, and it's modifying the state of the object which feels like a side effect. Furthermore if I pass some junk that responds to `to_s` here we'll just try to load that, and that doesn't seem safe. 

Ideally this should be directly supported by `YAML.load`, but short of that I would do this in `initialize` and more explicitly.

```ruby
@file_path = file_path.is_a?(Pathname) ? file_path.to_s : file_path
```

dblock commented on spec/hashie/parsers/yaml_erb_parser_spec.rb:
```diff
@@ -1,6 +1,21 @@
 require 'spec_helper'
 
 describe Hashie::Extensions::Parsers::YamlErbParser do
+  describe '#perform' do
```
There's already a `describe '.perform'` below, so combine them. Something like this:

```ruby
describe '.perform' do
   context 'a file' do
   end
   context 'Pathname' do
   end
end
```

dblock commented:
Please also squash your commits and update CHANGELOG and README.

gipcompany commented:
@dblock , I updated the code as you suggested but accidentally I deleted this pull request. Sorry. So I newly created another pull request on #337 .

Could you see the page?
